### PR TITLE
docs: Add event related documentation

### PR DESCRIPTION
**Pull request by bmtcril**
This moves in content from https://edx.readthedocs.io/projects/devdata that is not specific to the research packages. I removed old events which have been long deprecated and added in one new one that wasn't previously included.

### CODE REVIEW

File: source/developers/references/internal_data_formats/tracking_log_event_list.rst [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+Alphabetical Tracking Log Event List
+######################################
+
+
+:ref:`ABC` - :ref:`DEF` - :ref:`GHI` - :ref:`JKL` - :ref:`MNO`
+- :ref:`PQR` - :ref:`ST` - :ref:`UVWXYZ`
+
+.. _ABC:
+
+*******
```
Comment (feanil) [inline source/developers/references/internal_data_formats/tracking_log_event_list.rst:13]: NIT: I don't think the overlines here serve much of a purpose but fine to leave them if you like them stylistically.

File: source/developers/references/internal_data_formats/tracking_logs/common_fields.rst [hunk 2]
```diff
@@ -0,0 +1,259 @@
+
+.. _common:
+
+*****************************
```
Comment (feanil) [inline source/developers/references/internal_data_formats/tracking_logs/common_fields.rst:4]: Is there a reason this didn't start with hashes for the page title?

**Review by feanil — COMMENTED** [review]
Content wise this looks great, style wise, The overlines seem like a lot to me and generally we haven't used them much in the rest of this site yet so I'd love to avoid them unless you feel strongly about them.  I'm happy for that cleanup to be a follow on PR if that's easier to think about also.

File: source/developers/references/internal_data_formats/tracking_logs/common_fields.rst (hunk 2 above)
Comment (bmtcril) [reply source/developers/references/internal_data_formats/tracking_logs/common_fields.rst]: Just copied in that way, fixed now

**Review by feanil — APPROVED** [review]

**Pull request merged**